presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java

```diff
@@ -40,9 +40,14 @@
 
     public JdbcConnectorFactory(String name, Module module)
     {
-        this(name, JdbcModuleProvider.withCredentialProvider(module));
+        this(name, catalogName -> module);
     }
 
+    /**
+     * Prefer {@link JdbcConnectorFactory#JdbcConnectorFactory(String, Module)}.
+     * Notice that {@link io.prestosql.plugin.base.CatalogName} is available in guice context.
+     */
+    @Deprecated
```

Comment:
this is still _the constructor_ (with the other one delegating here).

Will we have this deprecated staying here & being used for ever?

Reply:
Once we start to use the other constructor widely, we could remove this one and then we could refactor this class and remove `JdbcModuleProvider`.